src/sentry/buffer/redis.py

```diff
@@ -323,20 +341,28 @@ def process_pending(self, partition=None):
         finally:
             client.delete(lock_key)
 
-    def process(self, key=None, batch_keys=None):
+    def process(self, key: str | None = None, batch_keys: list[str] | None = None) -> None:  # type: ignore[override]
```

Comment:
heh -- nothing to fix here but this `override` is actually pretty bad and why this module is so damn hard to understand what it does

the signature of this is completely different from what the "api" is expected of it and really makes everything very complicated

Reply:
100% agree